src/gameComponents/HullZone.java

```diff
@@ -0,0 +1,148 @@
+     * @param rightArcOffset .
+     * @param rightArcAngle .
+     */
+
+    public HullZone(Facing facing, int redBattery, int blueBattery,int blackBattery, int shields, Vector2f LoSOffset, Vector2f leftArcOffset, float leftArcAngle, Vector2f rightArcOffset, float rightArcAngle){
+        this.facing = facing;
+        this.redBattery = redBattery;
+        this.blueBattery = blueBattery;
+        this.blackBattery = blackBattery;
+        this.shields = shields;
+        this.shieldsMaximum = shields;
+        this.LoSOffset = LoSOffset;
+        this.leftArcOffset = leftArcOffset;
+        this.leftArcAngle = leftArcAngle;
+        this.rightArcAngle = rightArcAngle;
+        this.rightArcOffset = rightArcOffset;
+    }
+
+    /**
+     * Handles taking shield damage from an attack
+     * @param toTake the damage the hull zone is to take
+     * @return damage suffered in excess of shields. this should go to hull damage
+     */
+    public int sufferDamage(int toTake) {
+        if (toTake > shields) { //If we need to take more damage than shields remaining
+            shields = 0;
+            return (toTake - shields);
+        } else if (toTake < shields) { //if we need to take less damage than shields remaining
+            shields -= toTake;
+        } else shields = 0; //if we reach this point we are taking exactly as much damage as shields remaining
+        return 0; //in cases 2 and 3, all damage has been mitigated and nothing needs ot be suffered on hull, so this lives outside the if/else blocks
+    }
+
+    /**
+     * How a ship gains shields, such as from card effects or spending engineering points
+     * @param toGain the number of shields a hull zone will attempt to gain
+     * @return the number of shields actually gained
+     */
+    public int gainShield(int toGain){
+        int shieldsRemaining = Math.min(shields+toGain,shieldsMaximum);
+        int shieldsToGain  =  shieldsRemaining-shields;
+        shields = shieldsRemaining;
+        return shieldsToGain;
+    }
+
+    /**
+     * How a ship loses shields, such as from card effects or spending engineering points
+     * @param toRemove the number of shields a hull zone will attempt to remove
+     * @return the number of shields actually removed
+     */
+    public int removeShield(int toRemove){
+        int shieldsRemaining = Math.max(shields-toRemove,0);
+        int shieldsToRemove = shields-shieldsRemaining;
+        shields = shieldsRemaining;
+        return shieldsToRemove;
+    }
+
+    public Facing getFacing() {
+        return facing;
+    }
+
+    public int getRedBattery() {
+        return redBattery;
+    }
+
+    public int getBlueBattery() {
+        return blueBattery;
+    }
+
+    public int getBlackBattery() {
+        return blackBattery;
+    }
+
+    public int getShields() {
+        return shields;
+    }
+
+    public Vector2f getLoSOffset() {
+        return LoSOffset;
+    }
+
+    public Vector2f getLeftArcOffset() {
```

Comment:
doing a getter for objects like this can be counter-productive to making the field private.  An outside class could do thingy = hullZoneGuac.getLeftArcOffset()
thingy.changeX().

Now hullZoneGuacs private object has been altered by someone else on accident.

If you need to use this sort of thing, return a copy of the object, not the object, and NOT a reference to the same object.  Always good to test to make sure you are returning a copy and not the actual thing.

Reply:
Refactors will clean up this particular one.